db.go

```diff
@@ -448,11 +451,34 @@ func (db *DB) recover(ctx context.Context, wal WAL) error {
 		)
 		snapshotTx = 0
 	}
-	if snapshotTx != 0 {
+	firstIndex, err := wal.FirstIndex()
+	if err != nil {
 		level.Info(db.logger).Log(
-			"msg", "successfully loaded snapshot",
-			"tx", snapshotTx,
-			"duration", time.Since(snapshotLoadStart),
+			"msg", "failed to get WAL first index",
+			"err", err)
+	}
+	lastIndex, err := wal.LastIndex()
+	if err != nil {
+		level.Info(db.logger).Log(
+			"msg", "failed to get WAL last index",
+			"err", err)
+	}
+	snapshotLogArgs := make([]interface{}, 0)
```

Comment:
Isn't this `any` over `interface` now?
```suggestion
	snapshotLogArgs := make([]any{}, 0)
```

Reply:
Fair point, will change